examples/ajax/src/Main.purs

```diff
@@ -19,91 +14,76 @@ import Halogen.HTML.Properties as HP
 import Halogen.Util (runHalogenAff, awaitBody)
 import Halogen.VirtualDOM.Driver (runUI)
 
-import Network.HTTP.Affjax (AJAX, post)
+import Network.HTTP.Affjax as AX
 
--- | The state of the component.
-type State = { busy :: Boolean, code :: String, result :: Maybe String }
-
-initialState :: State
-initialState = { busy: false, code: exampleCode, result: Nothing }
-
-exampleCode :: String
-exampleCode = """module Main where
-
-import Prelude
-import Control.Monad.Eff.Console (print)
-
-fact :: Int -> Int
-fact 0 = 1
-fact n = n * fact (n - 1)
-
-main = print (fact 20)
-"""
+-- | The component state record.
+type State =
+  { busy :: Boolean
+  , username :: String
+  , result :: Maybe String
+  }
 
 -- | The component query algebra.
 data Query a
-  = SetCode String a
-  | MakeRequest String a
+  = SetUsername String a
+  | MakeRequest a
 
--- | The effects used in the app.
-type AppEffects eff = H.HalogenEffects (ajax :: AJAX | eff)
+-- | The effects used within the component.
+type Effects eff = (ajax :: AX.AJAX | eff)
 
--- | The definition for the app's main UI component.
-ui :: forall eff. H.Component HH.HTML Query Void (Aff (AppEffects eff))
-ui = H.component { render, eval, initialState }
-  where
-
-  render :: State -> H.ComponentHTML Query
-  render st =
-    HH.div_ $
-      [ HH.h1_
-          [ HH.text "ajax example / trypurescript" ]
-      , HH.h2_
-          [ HH.text "purescript input:" ]
-      , HH.p_
-          [ HH.textarea
-              [ HP.value st.code
-              , HE.onValueInput (HE.input SetCode)
-              ]
-          ]
-      , HH.p_
-          [ HH.button
-              [ HP.disabled st.busy
-              , HE.onClick (HE.input_ (MakeRequest st.code))
-              ]
-              [ HH.text "Compile" ]
-          ]
-      , HH.p_
-          [ HH.text (if st.busy then "Working..." else "") ]
-      ]
-      <> flip foldMap st.result \js ->
-          [ HH.div_
-              [ HH.h2_
-                  [ HH.text "javascript output:" ]
-              , HH.pre_
-                  [ HH.code_ [ HH.text js ] ]
-              ]
-          ]
-
-  eval :: Query ~> H.ComponentDSL State Query Void (Aff (AppEffects eff))
-  eval (SetCode code next) = H.modify (_ { code = code, result = Nothing :: Maybe String }) $> next
-  eval (MakeRequest code next) = do
+-- | The state value to use when the component is initialized.
+initialState :: State
+initialState = { busy: false, username: "", result: Nothing }
+
+-- | The component renderer.
+render :: State -> H.ComponentHTML Query
+render st =
+  HH.div_ $
+    [ HH.h1_ [ HH.text "Lookup GitHub user" ]
+    , HH.label_
+        [ HH.div_ [ HH.text "Enter username:" ]
+        , HH.input
+            [ HP.value st.username
+            , HE.onValueInput (HE.input SetUsername)
+            ]
+        ]
+    , HH.button
+        [ HP.disabled st.busy
+        , HE.onClick (HE.input_ MakeRequest)
+        ]
+        [ HH.text "Fetch info" ]
+    , HH.p_
```

Comment:
Yeah, I'd probably do it a bit differently for a real thing, but I want to keep everything super simple that is not related to what the example is trying to demonstrate - I'd probably `foldMap` or `guard` the things in a real situation.